### ISSUE TITLE
Upgrade OpenMage : custom theme not work (solved)

### ISSUE THREAD

**Issue by Danieleeffe1** [OpenMage/magento-lts#1796]
### Preconditions (*)

1. openmage 20.0.8
2. ubuntu server 20.04
3. php 7.4
4. apache 2
5. mysql 8

### Steps to reproduce (*)

1. upgrade openmage 20.0.8 to 20.0.12 from composer 2.0

### Expected result (*)

1. normal upgrade

### Actual result (*)
0. The website not load custom file Style.css 
1. I try test assignin default theme openmage front end this works, but if i change in custom theme the file css not work .
2. The css custom theme have a permission correct but the file not loaded
3. if i see in frontpage the html in the head the link "stylesheet" of the theme custom (style.css) file does not exist.
3. The custom theme is MIMIC (openmage theme section)

**Comment by luigifab** [comment]
Perhaps it is same as #1748, from #1073. My bad.

**Comment by ADDISON74** [comment]
You can do a comparison between 20.0.8 and 20.0.12 here https://github.com/OpenMage/magento-lts/compare/v20.0.8...v20.0.12.

Going through the list of commits I did not find anything that could affect your installation. Maybe just this one: **Removed "default" themes #1600**.

I would install 20.0.8 and 20.0.12 from scratch in two different directories without any dump file then I would install the custom theme in both and do the checks. Only at that moment if you can reproduce the problem in 20.0.12 it means that something happened between the versions. Otherwise, it could be an incomplete upgrade.

**Comment by pauldpauld** [comment]
Hi,
I recently installed a full 20.0.12 using composer, not an upgrade.
I am experiencing the same difficulity, the RWD theme works fine, trying to change to any uploaded theme (with correct ownership and permissions), gives the same error of missing css.
Thanks, Paul

**Comment by Danieleeffe1** [comment]
Maybe I have the temporary solution, go into categories and assign the custom style to the root category (or desired categories)

**Comment by Danieleeffe1** [comment]
Here we go again, same problem

upgrade from 20.0.12 to 20.0.13

custom fears does not work, you lose the css.

Can you check for this bug that occurs every single time you update openmage?

the custom theme and a variant of rwd (Mimic) where there are only the files indicated in the repositories (still under control) present in OpenMage theme.

this time I also created a completely new thema folder independent of rwd, result: it doesn't work

**Comment by Danieleeffe1** [comment]
> Forse è lo stesso di #1748 , da #1073 . Colpa mia.

Hello

I read a little about the reasons for these changes.

I would like to specify that with no browser the css file of the customized theme is loaded

the custom theme is a direct descendant of RWD

apart from the css file and a few modifications of some files that do not affect the classes in any way.

the result is that themes other than rwd are not loaded.

it's a huge problem

ps: the problem recurs also with the 20.0.13 version of openmage

**Comment by ADDISON74** [comment]
I am interested in investigating this issue but I need to describe in detail all the your step in reproducing it after installing from scratch OpenMage 20.0.13, including any modification in website settings. I'm sorry to comment, but the way you reported this issue is incomplete, even vaguely.

**Comment by Danieleeffe1** [comment]
thank you for your interest.

I use google translate.

I apologize for any communication difficulties.

it's been a few months so I'll tell the installation by heart

I created a server:

ubuntu 20.04
mysql 8
apache 2
php 8.x (active)
php 7.4 (active via htaccess for openmage)
php7.2
php 5.6

i have the following installation path of the websites

/ var / www /

openmage is installed in the subdirectori

/ var / www / openmage

I transferred the openmage source to the openmage folder.

as the composer installs openmage in a subdirectory with assigned name I had to copy the composer.json file in the "www" directory

script of composer.json:
`{
    "require": {
        "aydin-hassan/magento-core-composer-installer": "*",
        "openmage/magento-lts": "*"
    },
    "extra": {
        "magento-core-package-type": "magento-source",
        "magento-root-dir": "openmage"
    }
}`

I assigned a "magento" user with read-write permissions to the openmage folder and the composer.json file

via ssh as user "magento" I started the command "composer update"

Composer ran its processes without errors by installing OpenMage 20.0.8 , all this to be able to upgrade OpenMage without errors via Composer in the future. 

about 2 weeks ago I updated magento from 20.0.8 to 20.0.12 via "composer update" command, composer performed everything without errors.

since that moment the custom template has not worked anymore

the template can be found here:
[https://github.com/OpenMage/OpenMage-Themes/pull/1](Mimic OpenMage).

I have brought OpenMage back to version 20.0.8 with a backup file and it worked

I tried 2 days ago to update OpenMage to version 20.0.13 but it has the same problem as 20.0.12

that's all I remember doing.

**Comment by ADDISON74** [comment]
You practically have a customized version and after an update you came to the conclusion that part of it no longer works. 

The reproduction had to be done by installing the OpenMage 20.0.8 - 20.0.13 versions in different directories together with the Mimic theme. Then make the configuration in each installation to use this theme. If a malfunction is found in version 20.0.13 then it is a problem and it is worth investigating here.

This is what I am going to do. I will install all those versions to reproduce what you reported in the first post and I will come back with a feedback here.

**Comment by ADDISON74** [comment]
There is also the possibility that the upgrade you made using the composer was not complete. Personally I am using a manual upgrade, I download a copy of the latest Openmage in a directory then I use a script that completely deletes all the files installed by the previous version and then copies the new ones. I also use git clones update OM in my test environment.

**Comment by ADDISON74** [comment]
The fact that you use Google Translate should not have been a problem in reporting all the steps you followed in reproducing the problem. Unfortunately, you did not provide me with the information I wanted..

Here is how I tried to reproduce the problem you reported initially:

1. I downloaded all the official OpenMage archives from the GitHub page 20.0.8, 20.0.10, 20.0.11, 20.0.12, 20.0.13 each in its own directory.

2. I created 5 databases for each installation. I unzipped the archives and proceeded to manual installation. Except for version 20.0.8, which lacks the LICENSE.html file, the rest of the installations went smoothly.

3. In Backend I disabled all the options in the Cache Management section. I downloaded the MIMIC theme from the page where you uploaded it and I copied the app and skin directories in all 5 installations.

4. In Backend I went to the Settings section then to the General -> Design tab and I filled in Translations, Templates, Skin, Layout, Default with mimic word. Thus the first search starts in the rwd package and then the mimic directory.

5. I loaded Frontend for each installation and then looked in the source code of the Home page because you reported that it does not load the custom style.css files.

OpenMage 20.0.8: I did not identify any issues
OpenMage 20.0.10: I did not identify any issues
OpenMage 20.0.11: **I identified an issue. Two important css files are not loaded**
OpenMage 20.0.12: the same issue as above
OpenMage 20.0.13: the same same issue as above

If we make a difference between Home pages for 20.0.10 and 20.0.11 versions we find that in 20.0.11 the following lines at 17 and 53 are missing in the first part:
```
<!--[if lt IE 7]>
<script type="text/javascript">
//<![CDATA[
    var BLANK_URL = 'http://www.mydomain.tld/openmage/10/magento/js/blank.html';
    var BLANK_IMG = 'http://www.mydomain.tld/openmage/10/magento/js/spacer.gif';
//]]>
</script>
<![endif]-->
```

```
<!--[if  (lte IE 8) & (!IEMobile)]>
<link rel="stylesheet" type="text/css" href="http://www.mydomain.tld/openmage/10/magento/skin/frontend/rwd/mimic/css/styles-ie8.css" media="all" />
<link rel="stylesheet" type="text/css" href="http://www.mydomain.tld/openmage/10/magento/skin/frontend/rwd/mimic/css/madisonisland-ie8.css" media="all" />
<![endif]-->
<!--[if (gte IE 9) | (IEMobile)]><!-->
<link rel="stylesheet" type="text/css" href="http://www.mydomain.tld/openmage/10/magento/skin/frontend/rwd/mimic/css/styles.css" media="all" />
<link rel="stylesheet" type="text/css" href="http://www.mydomain.tld/openmage/10/magento/skin/frontend/rwd/mimic/css/madisonisland.css" media="all" />
<!--<![endif]-->
```

In the image below you can see the source code of the Home page and the last lines that contain CSS files are missing starting with version 20.0.11.

![20 0 10](https://user-images.githubusercontent.com/8360474/132134813-683c192d-f25b-4758-ace0-818cb4d368d6.jpg)

I went to version 20.0.11 and deleted in the Design tab all the 5 input boxes that contain the word mime. In Frontend the Home page loaded without issues in the layout.

**In conclusion, I confirm this major problem in OpenMage**. It is shown that between 20.0.10 and 20.0.11 there was a issue that affects the loading of the CSS files that are responsible with the layout. In my opinion, the removal of lines of code related to the Internet Explorer browser has led to this situation. The person who reported this issue mentioned that it consisted of an upgrade on 20.0.12 but in fact the issue appeared on 20.0.11. 

Obviously in the past although it was considered that removing lines from the template is strictly for Internet Explorer it also affects other browsers. I used in my tests Chromium, Firefox, Vivaldi and Brave. More attention needs to be paid to such situations because a lot of PR's have been created in the last year and even merged into the OpenMage code. Maybe we should review them as soon as possible.

**Comment by ADDISON74** [comment]
After investigating in more detail I came to the conclusion that the Mimic theme uses layout files that have not been updated with the latest RWD version which has changes in 20.0.11.

If the /app/design/frontend/rwd/mimic/layout/page.xml file is renamed then the one in /app/design/frontend/rwd/default/layout/page.xml will be used and the CSS files are loaded. In this way the page doesn't have any issues with the layout. You can see below the proof that once the XML file name was changed, the styles.css file from mimic directory was loaded.

![mimic](https://user-images.githubusercontent.com/8360474/132136166-407c0fda-197f-4a28-80b7-1a0a24cbfa93.jpg)

In conclusion, this issue report is not coming from OpenMage code but in the theme files used. Therefore it can be closed. Please in the future that those who report issues to perform tests to allow us to start from a level not from the ground.

**Comment by Danieleeffe1** [comment]
thank you all for the inquiry and time commitment.

I will modify Mimic to make it compatible.  on this occasion I will write a
small tutorial for those who in the future have a custom thema based on RWD
prior to 20.0.8

Il dom 5 set 2021, 19:41 ADDISON ***@***.***> ha scritto:

**Comment by Danieleeffe1** [comment]
> Dopo aver indagato più in dettaglio, sono giunto alla conclusione che il tema Mimic utilizza file di layout che non sono stati aggiornati con l'ultima versione RWD che presenta modifiche in 20.0.11.
> 
> Se il file /app/design/frontend/rwd/mimic/layout/page.xml viene rinominato, verrà utilizzato quello in /app/design/frontend/rwd/default/layout/page.xml e i file CSS vengono caricati . In questo modo la pagina non ha problemi con il layout. Di seguito puoi vedere la prova che una volta modificato il nome del file XML, è stato caricato il file styles.css dalla directory mimic.
> 
> ![imitare](https://user-images.githubusercontent.com/8360474/132136166-407c0fda-197f-4a28-80b7-1a0a24cbfa93.jpg)
> 
> In conclusione, questo rapporto sul problema non proviene dal codice OpenMage ma dai file del tema utilizzati. Quindi può essere chiuso. In futuro chi segnala problemi è pregato di eseguire dei test per permetterci di partire da un livello non da terra.

the solution you suggest is applicable only and exclusively if the custom theme is limited to being nested in RWD.

If a user wants to create a separate Theme folder from RWD the problem persists.

If there is a guideline informing web designers how to modify templates prior to opemage version 20.0.11 please leave the link.

Thanks.

**Comment by Danieleeffe1** [comment]
Resolved

I cloned the RWD certella both in app / design / and in Skin / frontend

I changed the RWD name to the custom template name

I then copied the custom theme files without selecting the files:

/app/design/frontend/tema-custom/etc/theme.xml

And

/app/design/frontend/tema-custom/layout/page.xml


now works
